addons/portal/models/mail_message.py

```diff
@@ -94,10 +95,13 @@ def _portal_message_format(self, properties_names, options=None):
             if message_to_attachments:
                 values['attachment_ids'] = message_to_attachments.get(message.id, {})
             if 'author_avatar_url' in properties_names:
-                if options and 'token' in options:
-                    values['author_avatar_url'] = f'/mail/avatar/mail.message/{message.id}/author_avatar/50x50?access_token={options["token"]}'
-                elif options and options.keys() >= {"hash", "pid"}:
-                    values['author_avatar_url'] = f'/mail/avatar/mail.message/{message.id}/author_avatar/50x50?_hash={options["hash"]}&pid={options["pid"]}'
+                if message.author_id:
```

Comment:
What's the point of this extra conditions if the `else` case is  displaying the same kind of URL anyway.

Reply:
In case of having guest (if we delete the partner_id of the document) there is still a token on the page, but the author would be guest and the image URL would be `/mail/avatar/mail.message/message_id/author_avatar/50x50?access_token=some token` which is visually like `/web/image/mail.message/message_id/author_avatar/50x50` because there is no access for guest. But they're different URLs. So I put the condition of checking the author of the message to return the more proper avatar if the author is a guest.  
As discussed, it's not what we want here, and it should be correct in another PR later.